app/assets/javascripts/discourse/app/modifiers/post-stream-viewport-tracker.js

```diff
@@ -642,7 +688,12 @@ export default class PostStreamViewportTracker {
 
     this.#cloakingObserver = this.#initializeObserver(this.trackCloakedPosts, {
       rootMargin: `${this.#cloakOffset}px 0px`,
-      threshold: [0, 1],
+      // Adding UNCLOAKING_HYSTERESIS_RATIO provides a threshold between completely cloaked (0) and fully visible (1)
+      // states. Without this threshold, the intersection observer would only trigger when the element entered or exited
+      // the cloaking viewport. The values are clamped between 0 and 1.
+      threshold: Array.from(new Set([0, UNCLOAKING_HYSTERESIS_RATIO, 1]))
+        .map((n) => Math.max(0, Math.min(n, 1)))
+        .sort(),
```

Comment:
Is all this code to ensure we never set a value for `UNCLOAKING_HYSTERESIS_RATIO` which would create a duplicate value or change the sorting?

because it feels like we are just doing: `[0, UNCLOAKING_HYSTERESIS_RATIO, 1]` here